src/Value/Container.php

```diff
@@ -118,4 +128,26 @@ protected function setTraverse($key, $value)
         $ref = $value;
         return $this;
     }
+
+    /**
+     * @param array $keyParts
+     * @param array $values
+     */
+    private function removeRecursive(array $keyParts, array &$values)
+    {
+        $key = $keyParts[0];
+        if (array_key_exists($key, $values)) {
+            if (is_array($values[$key])) {
+                $this->removeRecursive(array_splice($keyParts, 1), $values[$key]);
+
+                // unset self if the removed child clears this array
```

Comment:
Tried that, but for some reasons that breaks the reference to the object, and the unset will fail